lib/filestorage.js

```diff
@@ -160,9 +147,9 @@ class FileStorage {
 
   // Start compressing stashed files
   //   cb - <Function>, callback to be called on each finished compression
+  // Throws: <TypeError>, if compression is incorrect
   startCompressing(cb) {
-    while (this.compressionList.length) {
-      const file = this.compressionList.unshift();
+    for (const file of this.compressionList) {
       this._compress(file.id, file.compression, cb);
     }
```

Comment:
You should `this.compressionList.length = 0` or `this.compressionList = []` as this was the previous behavior if it's needed.